src/EditorFeatures/Core/Shared/Utilities/VisualStudioTaskSchedulerProvider.cs

```diff
@@ -2,28 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Collections.Generic;
 using System.Composition;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
 {
-    [ExportWorkspaceService(typeof(ITaskSchedulerProvider), ServiceLayer.Host), Shared]
+    [ExportWorkspaceService(typeof(ITaskSchedulerProvider), ServiceLayer.Editor), Shared]
     internal sealed class VisualStudioTaskSchedulerProvider : ITaskSchedulerProvider
```

Comment:
This should probably be renamed?

Reply:
I kept the name because it's still tied to Microsoft.VisualStudio.Threading.JoinableTaskContext.

Reply:
JoinableTaskContextTaskSchedulerProvider?

Reply:
Does `ThreadingContextTaskSchedulerProvider` work?

Reply:
I like that too.